test/dynamo/test_export.py

```diff
@@ -3286,18 +3286,14 @@ def forward(self, pred, x):
         cos = arg1.cos();  arg1 = None
         return pytree.tree_unflatten([cos], self._out_spec)
 """
-        true_guard_code = ["cast_symbool_to_symint_guardless(L['pred']) == 1"]
-        false_guard_code = [
-            "Ne(cast_symbool_to_symint_guardless(L['pred']), 1)",
-            "-9223372036854775808 <= cast_symbool_to_symint_guardless(L['pred'])",
-        ]
+        true_guard_code = ["ITE(L['pred'], 1, 0) == 1"]
+        false_guard_code = ["Ne(ITE(L['pred'], 1, 0), 1)"]
         test_symbool_guards(
             f,
             [3, 3, 4, 5],
             [true_graph, true_graph, false_graph, false_graph],
             [true_guard_code, true_guard_code, false_guard_code, false_guard_code],
-            # Outter shape env should have no guards in it because we never specialize on the outter symbool.
-            [[], [], [], []],
+            [["Eq(s0, 3)"], ["Eq(s0, 3)"], ["Eq(s0, 3)"], ["Eq(s0, 3)"]],
```

Comment:
Why are the false cases also guarding Eq(s0, 3)? It doesn't seem like s0 will equal 3 here?

Reply:
This is because consecutive export's sharing the same shape_env. So the pred for each export is actually:

  1. Eq(s0, 3)
  2. Eq(s0, 3), which is evaluated to True
  3. Eq(s0, 4), which is evaluated to False
  4. Eq(s0, 5), which is evaluated to False

When propagating the following guards created by consecutive export:

1. ["ITE(L['pred'], 1, 0) == 1"] 
2. ["ITE(L['pred'], 1, 0) == 1"] 
3. ["Ne(ITE(L['pred'], 1, 0), 1)"]
4. ["Ne(ITE(L['pred'], 1, 0), 1)"] 

Only 1. will actually propagate the Eq(s0, 3) guard to outer shape_env, the rest will be True automatically due to the Ea(s0, 3) guard.